sorting/merge_sort.c

```diff
@@ -33,6 +33,11 @@ void swap(int *a, int *b)
 void merge(int *a, int l, int r, int n)
 {
     int *b = (int *)malloc(n * sizeof(int)); /* dynamic memory must be freed */
+    if (b == NULL) /* return to main if can't malloc memory */
```

Comment:
Removes this outdated comment since the program exits now
```suggestion
    if (b == NULL)
```